src/components/screens/vacation-requests-screen.tsx

```diff
@@ -200,6 +199,67 @@ const VacationRequestsScreen = () => {
     setLoading(false);
   };
 
+  /**
+   * Update a vacation request status
+   *
+   * @param status VacationRequestStatus value for the new status
+   * @param selectedRowIds GridRowId[] list of Ids of selected rows(vacation requests)
+   */
+  const updateVacationRequestStatus = async (
+    status: VacationRequestStatuses,
+    selectedRowIds: GridRowId[]
+  ) => {
+    if (!loggedInUser) return;
+
+    try {
+      setLoading(true);
+      const updatedVacationRequests = await Promise.all(
+        selectedRowIds.map(async (vacationRequestId) => {
+          const vacationRequest = vacationRequests.find(
+            (vacationRequest) => vacationRequest.id === vacationRequestId
+          );
+          if (!vacationRequest) return;
+
+          const statusExists = vacationRequest.status?.some(
+            (existingStatus) => existingStatus.createdBy === loggedInUser.id
+          );
+
+          const newOrUpdatedStatus = {
+            status,
+            createdBy: loggedInUser.id,
+            updatedAt: new Date(),
+            message: `This vacation request is ${status}`
+          };
+
+          const updatedStatus = statusExists
+            ? vacationRequest.status?.map((existingStatus) =>
+                existingStatus.createdBy === loggedInUser.id
+                  ? { ...existingStatus, ...newOrUpdatedStatus }
+                  : existingStatus
```

Comment:
If existingStatus is true don't we know that it was created by the logged in user? In which case this inner ternary becomes redundant?

Also If there is an existing status from the current user don't we want to update it/ replace it, rather than just add another status on top of it { ...existingStatus, ...newOrUpdatedStatus }? And in the Or case, if the existing status wasn't created by the currently updating user we are now replacing it : existingStatus, so replacing a status created by someone else. This also doesn't seem right. Perhaps these conditions are the wrong way round? Perhaps I'm mistaken, lets discuss further if needed